src/EditorFeatures/Core/IntelliSense/AsyncCompletion/ItemManager.CompletionListUpdater.cs

```diff
@@ -238,33 +252,36 @@ private void AddCompletionItems(List<MatchResult> list, CancellationToken cancel
                 // FilterStateHelper is used to decide whether a given item should be included in the list based on the state of filter/expander buttons.
                 var filterHelper = new FilterStateHelper(_snapshotData.SelectedFilters);
 
-                using var _1 = PooledHashSet<string>.GetInstance(out var includedPreferredItems);
-                using var _2 = ArrayBuilder<MatchResult>.GetInstance(out var includedDefaults);
-                var unmatchedDefaults = _snapshotData.Defaults;
+                var includedPreferredItems = new ConcurrentSet<string>();
+                var includedDefaults = new ConcurrentDictionary<string, MatchResult>();
+
+                Enumerable.Range(0, _snapshotData.InitialSortedItemList.Count)
+                    .AsParallel()
+                    .WithCancellation(cancellationToken)
+                    .ForAll(CreateMatchResultAndProcessMatchingDefaults);
```

Comment:
curious why .ForAll, instead of .Where.  Then we don't need a lock, right?

Reply:
Yes, you are right. I saw in my local measurement ForAll has small (but largely negligible) edge over Where, that's why I pick ForAll.